backtracking/sudoku.py

```diff
@@ -51,24 +54,21 @@ def is_safe(grid: Matrix, row: int, column: int, n: int) -> bool:
         if grid[row][i] == n or grid[i][column] == n:
             return False
 
-    for i in range(3):
-        for j in range(3):
-            if grid[(row - row % 3) + i][(column - column % 3) + j] == n:
-                return False
-
-    return True
+    return all(
+        grid[(row - row % 3) + i][(column - column % 3) + j] != n
+        for i, j in itertools.product(range(3), range(3))
```

Comment:
```suggestion
        for i in range(3) for j in range(3))
```
This should do the same thing without needing an imported function